### PR TITLE
Restart a vm geneva action

### PR DESCRIPTION
**Pull request by gburges**
```release-note
NONE
```

### CODE REVIEW

**Comment by gburges** [comment]
@ehashman ptal, this pr looks better now

File: pkg/plugin/plugin.go [hunk 1]
```diff
@@ -525,6 +525,39 @@ func (p *plugin) ListClusterVMs(ctx context.Context, cs *api.OpenShiftManagedClu
 	return &api.GenevaActionListClusterVMs{VMs: &pods}, nil
 }
 
+func (p *plugin) Restart(ctx context.Context, cs *api.OpenShiftManagedCluster, hostname string) error {
+	if !validate.IsValidAgentPoolHostname(hostname) {
+		return fmt.Errorf("invalid hostname %q", hostname)
+	}
```
Comment (jim-minter) [inline pkg/plugin/plugin.go]: rebase and include a call to `checkIfClusterWillRefresh` here like the other geneva actions now do

**Comment by jim-minter** [comment]
have you manually tested that this PR works for master, infra and compute nodes?

File: pkg/plugin/plugin.go (hunk 1 above)
Comment (gburges) [reply pkg/plugin/plugin.go]: added

**Comment by gburges** [comment]
> have you manually tested that this PR works for master, infra and compute nodes?

Yes, works on any and all of them

**Comment by gburges** [comment]
/test upgrade-v7.1

timeout issue

**Comment by jim-minter** [comment]
@gburges now please work with tong or zhuoli (not sure which one) to get this enabled on MSFT's side.

**Pull request merged**